src/main/java/net/wasdev/wlp/common/plugins/util/DevUtil.java

```diff
@@ -0,0 +1,956 @@
+import java.util.List;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+import javax.tools.JavaCompiler;
+import javax.tools.JavaFileObject;
+import javax.tools.StandardJavaFileManager;
+import javax.tools.StandardLocation;
+import javax.tools.ToolProvider;
+
+import org.apache.commons.io.FileUtils;
+
+import com.sun.nio.file.SensitivityWatchEventModifier;
+
+/**
+ * Utility class for dev mode.
+ */
+public abstract class DevUtil {
+
+    /**
+     * Log debug
+     * 
+     * @param msg
+     */
+    public abstract void debug(String msg);
+
+    /**
+     * Log debug
+     * 
+     * @param msg
+     * @param e
+     */
+    public abstract void debug(String msg, Throwable e);
+
+    /**
+     * Log debug
+     * 
+     * @param e
+     */
+    public abstract void debug(Throwable e);
+
+    /**
+     * Log warning
+     * 
+     * @param msg
+     */
+    public abstract void warn(String msg);
+
+    /**
+     * Log info
+     * 
+     * @param msg
+     */
+    public abstract void info(String msg);
+
+    /**
+     * Log error
+     * 
+     * @param msg
+     */
+    public abstract void error(String msg);
+
+    /**
+     * Returns whether debug is enabled by the current logger
+     * 
+     * @return whether debug is enabled
+     */
+    public abstract boolean isDebugEnabled();
+
+    /**
+     * Stop the server
+     */
+    public abstract void stopServer();
+
+    /**
+     * Starts the default server
+     */
+    public abstract void startServer();
+
+    /**
+     * Updates artifacts of current project
+     */
+    public abstract void getArtifacts(List<String> artifactPaths);
+
+    /**
+     * Recompile the build file
+     * 
+     * @param buildFile
+     * @param artifactPaths
+     * @param executor The thread pool executor
+     * @return true if the build file was recompiled with changes
+     */
+    public abstract boolean recompileBuildFile(File buildFile, List<String> artifactPaths, ThreadPoolExecutor executor);
+
+    /**
+     * Get the number of times the application updated message has appeared in the application log
+     * 
+     * @return 
+     */
+    public abstract int countApplicationUpdatedMessages();
+
+    /**
+     * Run unit and/or integration tests
+     * 
+     * @param waitForApplicationUpdate Whether to wait for the application to update before running integration tests
+     * @param messageOccurrences The previous number of times the application updated message has appeared.
+     * @param executor The thread pool executor
+     * @param forceSkipUTs Whether to force skip the unit tests
+     */
+    public abstract void runTests(boolean waitForApplicationUpdate, int messageOccurrences, ThreadPoolExecutor executor, boolean forceSkipUTs);
```

Comment:
Could this method implementation also be moved to common?

Reply:
Addressed above, we can open an issue to track using Ant directly at the common level. Until that issue is resolved I think it is best to leave this as an abstract method for now.

Reply:
Moved to common in the following commits:
https://github.com/WASdev/ci.common/pull/48/commits/72a6d9ff57e101f0d34dd3771a49e03947bd9f32
https://github.com/WASdev/ci.maven/pull/476/commits/b51725037b22f1cc90ed46f11e1ec155966d1d6d